logstash-core/src/main/java/org/logstash/plugins/PluginFactoryExt.java

```diff
@@ -404,7 +405,7 @@ public Context toContext(PluginLookup.PluginType pluginType, AbstractNamespacedM
 
                     if (innerWriter != null) {
                         if (innerWriter.getJavaClass().equals(DeadLetterQueueWriter.class)) {
```

Comment:
I believe this would be a little less fragile (e.g., refactoring `DeadLetterQueueWriter.class`, specialised implementations etc.)

```suggestion
                         if (DeadLetterQueueWriter.class.isAssignableFrom(innerWriter.getJavaClass())) {
```